### ISSUE TITLE
Add "pay for order" ability to order confirmation

### ISSUE THREAD

**Issue by witoszekdev** [saleor/react-storefront#311]
We currently have an endpoint to pay for the order, but it still requires a payment method and payment provider to be passed from checkout. We need to figure out what would be the best way to do that

```[tasklist]
### Tasks
- [x] https://github.com/saleor/saleor/issues/11009
- [ ] Implement "pay for order" on the order confirmation page
- [ ] https://github.com/saleor/saleor/issues/11258
```

# Original issue:
Remove hardcoded payment gateway in `checkout-storefront`
Remove hardcoded payment gateway in [`src/sections/OrderInfo/PaymentSection.tsx:21`](https://github.com/saleor/saleor-checkout/blob/adyen-refunds/packages/checkout-storefront/src/sections/OrderInfo/PaymentSection.tsx#L21) and replace it with payment gateways fetched from app's metadata.

**Comment by mmiszy** [comment]
https://saleorcommerce.slack.com/archives/C039MGJKKHN/p1661260956559599

**Comment by mmiszy** [comment]
Report from user:
> Payment failure doesn't allow us to retry. What if I made a typo and wanted to re-enter the card number or want to try another card?
> Payment failures cause the checkout to no longer exist, changing the form causes errors, and reloading the page causes the cart to be empty.

Context: The problem occurs because we first create an order which makes the checkout token no longer valid... So when a payment fails, we cannot retry the process because checkout no longer exists and we don't store previously created orders. We need to change the architecture of how this works a little bit.
We should be able to fetch order by checkoutId.

**Comment by mmiszy** [comment]
Blocked by https://github.com/saleor/saleor/issues/11009

**Comment by mmiszy** [comment]
https://github.com/saleor/saleor/issues/11009

**Comment by mmiszy** [comment]
Blocked by https://github.com/saleor/saleor/issues/11258
